lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -3350,13 +3350,25 @@ LogicalResult ConvertAtenOp<PrimNumToTensorScalarOp>::matchAndRewrite(
   // Only supports integer operand type, because for the floating point operand
   // type result tensor has to be of type `f64` which is not supported in the
   // tosa.
-  int64_t initValue;
-  if (!matchPattern(op.getA(), m_TorchConstantInt(&initValue)))
-    return rewriter.notifyMatchFailure(
-        op, "unimplemented: input should be a torch constant int");
+  double doubleValue;
+  auto isFloat = matchPattern(op.getA(), m_TorchConstantFloat(&doubleValue));
+  int64_t intValue;
+  auto isInt = matchPattern(op.getA(), m_TorchConstantInt(&intValue));
+  if (!isFloat && !isInt)
+    return rewriter.notifyMatchFailure(op,
+                                       "Unable to extract the scalar constant");
+  float floatValue = static_cast<float>(doubleValue);
+
+  auto outElemTy = resultType.getElementType();
+  if (outElemTy.isa<mlir::IntegerType>() || outElemTy.isF32()) {
```

Comment:
`PrimNumToTensorScalarOp` does not produce `f32` tensors.

```python
In [5]: torch.ops.prim.NumToTensor(0.0).dtype
Out[5]: torch.float64
```

so this is not the right lowering for this op. Also, e2e tests are needed anytime a new op is extended. See: https://github.com/llvm/torch-mlir/blob/main/docs/architecture.md#dos-and-donts-for-unit-vs-end-to-end-testing

> DO use an [end-to-end test](https://github.com/llvm/torch-mlir/blob/main/docs/adding_an_e2e_test.md) if you are implementing a new op or extending the support for an existing op.

Reply:
Oh, it's XFAIL - "NumToTensorFloatModule_basic"

Reply:
I don't see this as having been addressed

Reply:
```
module attributes {torch.debug_module_name = "NumToTensorFloatModule"} {
  func.func @forward() -> !torch.vtensor<[],f64> {
    %float1.000000e00 = torch.constant.float 1.000000e+00 loc(#loc1)
    %0 = torch.prim.NumToTensor.Scalar %float1.000000e00 : !torch.float -> !torch.vtensor<[],f64> loc(#loc2)
    return %0 : !torch.vtensor<[],f64> loc(#loc)
  } loc(#loc)
} loc(#loc)
```
`➜  torch-mlir git:(indexput) ✗ torch-mlir-opt -convert-torch-to-tosa /tmp/NumToTensorFloatModule.mlir       `
```
 module attributes {torch.debug_module_name = "NumToTensorFloatModule"} {
  func.func @forward() -> !torch.vtensor<[],f64> {
    %float1.000000e00 = torch.constant.float 1.000000e+00
    %0 = "tosa.const"() {value = dense<1.000000e+00> : tensor<f32>} : () -> tensor<f32>
    %1 = "tosa.cast"(%0) : (tensor<f32>) -> tensor<f64>
    %2 = torch_c.from_builtin_tensor %1 : tensor<f64> -> !torch.vtensor<[],f64>
    return %2 : !torch.vtensor<[],f64>
  }
}
```

Reply:
Got it, Need to change the tosa::constOp to support Tosa_Tensor_Cast f64.
def Tosa_ConstOp : Tosa_Op<"const", ...
  let results = (outs
    Tosa_Tensor_Cast:$output
  );

Reply:
Done

Reply:
Done